feat: Implement LoginSuccessHandler

SafeDrivingPikachu opened the pull request:
- LoginSuccessHandler 구현
- KakaoUserInfo 구현
- Provider 수정 : of 메서드 구현(String 타입 provider를 ProviderType으로 변환하는 메서드)

- SocialProvider, SocialProviderRepository 수정사항 제외해야됨
- RefreshToken, RefreshTokenRepository 삭제예정이라고 합니다.

issue #14

sang218 committed:
fix(JwtManager): CreateRefreshToken, GetAuthentication

- JwtManager 수정:
    - CreateRefreshToken 메소드 수정: RefreshToken 객체 생성 추가
    - GetAuthentication 메소드 구현
- RefreshToken 수정:
    - 일반적으로 userId를 Id 값으로 사용한다고 해서 변경함
- RefreshTokenRepository 수정:
    - deleteByUserId 추가 (redis 토큰 삭제하는 역할)
- UserService 수정:
    - getAuthentication 구현
    - findUserById 구현 (현재 사용하지 않아서 주석 처리함)

- 테스트 필요

close #28, 29

sang218 committed:
fix: Implement UserDetails

- WebtyUserDetails 구현:
    - Spring Security 유저 정보 인터페이스 구현체
- WebtyUserDetailsService 구현:
    - Spring Security 유저 정보를 가져오는 인터페이스 구현체
- UserDetailsService 수정:
    - WebtyUserDetails 적용
- RedisConfig 수정:
    - RefreshTokenRepository 구현으로 필요없어진 RedisTemplate 주석 처리
- RefreshToken 심각한 오류 수정:
    - @Id import 잘못되어있어서 실행이 안되던 문제 해결
- CustomAuthenticationFilter 경미한 수정:
    - JwtManager import

- 테스트 필요

close #28, 29

sang218 committed:
fix: Move getAuthentication

- 메소드 위치 이동: userService -> JwtManager

close #28, 29

sang218 committed:
test: JwtManagerTest Implement

- 아직 돌려보지 못했습니다
- 테스트 필요

close #28, 29

sang218 committed:
chore: Merge branch 'main' of feature/webtyuesr/issue-29

SafeDrivingPikachu committed:
feat: Implement LoginSuccessHandler

- LoginSuccessHandler 구현
- KakaoUserInfo 구현
- Provider 수정 : of 메서드 구현(String 타입 provider를 ProviderType으로 변환하는 메서드)

- SocialProvider, SocialProviderRepository 수정사항 제외해야됨
- RefreshToken, RefreshTokenRepository 삭제예정이라고 합니다.

issue #14

DayDreaam commented on src/main/java/org/team14/webty/security/oauth2/LoginSuccessHandler.java:
```diff
@@ -1,4 +1,119 @@
+			// 신규 유저인 경우
+			log.info("신규 유저입니다. 등록을 진행합니다.");
+
+			socialProvider = SocialProvider.builder()
+				.provider(ProviderType.of(kakaoUserInfo.getProvider()))
+				.providerId(providerId)
+				.build();
+			socialProviderRepository.save(socialProvider);
+
+			user = WebtyUser.builder()
+				.nickname("nickname")
+				.socialProvider(socialProvider)
+				.build();
+			userRepository.save(user);
```
이 부분 UserService에서 add 함수 가져다 쓰셔도 될거같아요~

segye replied:
저희가 닉네임을 자동으로 지정해주는 부분이있어야해서 재현님 말씀처럼 UserService에서 add 함수를 사용하는게  맞는 것 같습니다!

sang218 replied:
반영했습니다

DayDreaam committed:
feature Webtoon 저장 구현 (KAKAO_PAGE, NAVER)

DayDreaam committed:
chore Transactional

Commit:
Merge pull request #49 from Dia218/feature/webtoon/issue-32

feature: Webtoon 저장 구현 (KAKAO_PAGE, NAVER)

sang218 committed:
chore: Merge branch 'main' into feature/webtyuesr/issue-29

sang218 committed:
fix: Delete Redis Repository

- redis 연결 방식을 기존 방식으로 되돌림
    - templete -> repository -> templete.

segye commented on src/main/java/org/team14/webty/security/oauth2/LoginSuccessHandler.java:
```diff
@@ -1,4 +1,119 @@
+
+		// 이름, 액세스 토큰, 리프레쉬 토큰을 쿠키에 담아 리다이렉트
+		String redirectUri = String.format(REDIRECT_URI);
+		response.addCookie(createCookie(accessToken, "accessToken"));
```
이 부분도 저희 commons/cookies에 있는 CookieManager 사용하셔도 될 것 같아요!

sang218 replied:
반영했습니다

sang218 committed:
fix: JwtManager GetAuthentication

- GetAuthentication 메소드가 작동하지 않던 오류를 수정했습니다
- JwtManager 에서 userId를 String 타입으로 사용하던 오류를 수정했습니다

- finish test (JwtManagerTest)

Commit:
Merge pull request #30 from Dia218/feature/webtyuesr/issue-29

fix(JwtManager): createRefreshToken & getAuthentication

Commit:
feat: Implement LoginSuccessHandler

- LoginSuccessHandler 구현:
    - 로그인에 성공했을 때 동작함
    - 회원 등록이 되어있지 않다면 등록하도록 UserService 호출
    - 토큰 생성 및 쿠키 생성 호출
- UserService 수정:
    - existSocialProvider 구현: providerId로 회원이 이미 등록되어있는지 확인 후 있으면 userId 반환
    - createUser 수정: 이름 변경 (add -> createUser) 및 리팩토링
- JwtManager 수정:
    - 리프레쉬 토큰 생성 시 redis에 저장하도록 수정

- ProviderUserInfo 구현:
    - 인증 정보에서 값을 추출할 때 사용하는 데이터 구조
- SocialProviderType 수정:
    - 이름 변경: ProviderType -> SocialProviderType
    - fromProviderName 추가: (String) 소셜로그인 제공자를 전달하면 그에 맞는 (enum)SocialProviderType 반환

- SocialProviderRepository 수정:
    - findByProviderId, @Repository 추가
- UserRepository 수정:
    - findBySocialProvider, @Repository 추가

sang218 commented on src/main/java/org/team14/webty/user/service/UserService.java:
```diff
@@ -18,47 +18,62 @@
 	private final SocialProviderRepository socialProviderRepository;
 	private final FileStorageUtil fileStorageUtil;
 
-	@Transactional
```
modifyNickname 삭제된 게 아니고 위치 아래로 이동했습니다

sang218 commented on src/main/java/org/team14/webty/user/service/UserService.java:
```diff
@@ -18,47 +18,62 @@
+		return webtyUser.getUserId();
 	}
 
-	@Transactional(readOnly = true)
```
findUserByNickname 완전히 삭제한 것은 아니고 위로 이동 후 주석처리 했습니다
사용하는 곳이 없어서요

sang218 reviewed (commented):
저도 이런적은 처음이라 잘 이해가 안가는데
이미 main에 있는 코드까지 새로 작성된 코드라고 뜹니다
원인을 모르겠습니다

sang218 reviewed (dismissed):
자동 병합이 되는지 확인

Commit:
fix: SocialProviderRepository Optional

- 해당 SocialProvider가 없을 때 처리하는 로직 추가

Commit:
fix: SecurityConfig, CustomAuthenticationFilter

- SecurityConfig 수정:
    - securityFilterChain: 로그인 시에만 접속 가능한 페이지 설정, 유효하지 않는 페이지 설정 (로그인하도록 유도 -> 접속 불가)
    - webSecurityCustomizer 구현: 스프링 시큐리티를 무시할 페이지 설정
- CustomAuthenticationFilter 수정:
    - shouldNotFilter: 스프링 시큐리티는 작동하되, 로그인 없이 접속 가능한 페이지 설정

sang218 reviewed (dismissed):
카카오로그인 테스트 완료했습니다!

검토하시면 병합해주세요

Dia218 commented:
확인 다 되신거겠죠...?
확인 하셨을 것 같아서 병합하겠습니다
